api/tests/unit/domain/usecases/retrieve-last-or-create-certification-course_test.js

```diff
@@ -468,97 +468,157 @@ describe('Unit | UseCase | retrieve-last-or-create-certification-course', functi
+                      habilitations: [],
+                      isV3Pilot: true,
+                    });
+                    certificationCenterRepository.getBySessionId.resolves(certificationCenter);
+
+                    certificationBadgesService.findStillValidBadgeAcquisitions
+                      .withArgs({ userId, domainTransaction })
+                      .resolves([]);
+
+                    const user = domainBuilder.buildUser({ id: userId });
+                    userRepository.get.withArgs(userId).resolves(user);
+
+                    const certificationCourseToSave = CertificationCourse.from({
+                      certificationCandidate: foundCertificationCandidate,
+                      challenges: [],
+                      verificationCode,
+                      maxReachableLevelOnCertificationDate: MAX_REACHABLE_LEVEL,
+                      version: 3,
+                      lang: user.lang,
+                    });
+
+                    const savedCertificationCourse = domainBuilder.buildCertificationCourse(
+                      certificationCourseToSave.toDTO(),
+                    );
+
+                    certificationCourseRepository.save
+                      .withArgs({ certificationCourse: certificationCourseToSave, domainTransaction })
+                      .resolves(savedCertificationCourse);
+
+                    const assessmentToSave = new Assessment({
+                      userId: 2,
+                      certificationCourseId: savedCertificationCourse.getId(),
+                      state: Assessment.states.STARTED,
+                      type: Assessment.types.CERTIFICATION,
+                      isImproving: false,
+                      method: Assessment.methods.CERTIFICATION_DETERMINED,
+                    });
+
+                    const savedAssessment = domainBuilder.buildAssessment(assessmentToSave);
+                    assessmentRepository.save
+                      .withArgs({ assessment: assessmentToSave, domainTransaction })
+                      .resolves(savedAssessment);
+
+                    // when
+                    const result = await retrieveLastOrCreateCertificationCourse({
+                      domainTransaction,
+                      sessionId: 1,
+                      accessCode: 'accessCode',
+                      userId,
+                      locale: 'fr',
+                      ...injectables,
+                    });
+
+                    // then
+                    expect(result).to.deep.equal({
+                      created: true,
+                      certificationCourse: new CertificationCourse({
+                        ...savedCertificationCourse.toDTO(),
+                        assessment: savedAssessment,
+                        challenges: [],
+                        version: 3,
+                        lang: user.lang,
+                      }),
+                    });
+                  });
```

Comment:
On pourrait aussi faire:
```suggestion
          const {created, certificationCourse} = await retrieveLastOrCreateCertificationCourse({
                      domainTransaction,
                      sessionId: 1,
                      accessCode: 'accessCode',
                      userId,
                      locale: 'fr',
                      ...injectables,
                    });

                    // then
                    expect(created).to.be.true;
                    expect(certificationCourse)to.deep.equalInstanceOf(new CertificationCourse({
                        ...savedCertificationCourse.toDTO(),
                        assessment: savedAssessment,
                        challenges: [],
                        version: 3,
                        lang: user.lang,
                      }),
                  });
```